openedx/features/subscriptions/models.py

```diff
@@ -0,0 +1,49 @@
+import logging
+
+from datetime import date
+from django.contrib.auth.models import User
+from django.db import models, transaction
+from model_utils.models import TimeStampedModel
+
+from student.models import CourseEnrollment
+
+class UserSubscription(TimeStampedModel):
+    """
+    The "UserSubscription" model.
+    """
+    user = models.ForeignKey(User, on_delete=models.CASCADE)
+    subscription_id = models.PositiveIntegerField()
+    sku = models.CharField(
+        max_length=255,
+        null=True,
+        blank=True,
+        verbose_name="SKU",
+    )
+    expiration_date = models.DateField(default=None, null=True, blank=True)
+
+    LIMITED_ACCESS = 'limited-access'
+    FULL_ACCESS_COURSES = 'full-access-courses'
+    FULL_ACCESS_TIME_PERIOD = 'full-access-time-period'
+    LIFETIME_ACCESS = 'lifetime-access'
+
+    SUBSCRIPTION_TYPE_CHOICES = (
+        (LIMITED_ACCESS, 'Limited Access', ),
+        (FULL_ACCESS_COURSES, 'Full Access (Courses)', ),
+        (FULL_ACCESS_TIME_PERIOD, 'Full Access (Time period)', ),
+        (LIFETIME_ACCESS, 'Lifetime Access', ),
+    )
+
+    subscription_type = models.CharField(choices=SUBSCRIPTION_TYPE_CHOICES, max_length=30)
+    number_of_courses = models.PositiveIntegerField(null=True, blank=True)
+    course_enrollments = models.ManyToManyField(CourseEnrollment, blank=True)
+
+
+    class Meta(object):
+        unique_together = ('subscription_id', 'user')
+
+    @classmethod
+    def get_active_subscription(self):
+        """
+        Get currently active subscription.
+        """
+        return self.objects.filter(expiration_date__gte=date.today())
```

Comment:
expiration date should either be null(full time access) or >= today.

Reply:
does that mean that expired `UserSubscriptions` would have null as `expiration_date` ?

Reply:
no. I mean for those subscription which are not time bound in those we'd have null in `expiration_date`.

Reply:
Oh got it. let me update it